better-click-to-tweet.php

```diff
@@ -223,31 +222,79 @@ function bctt_scripts() {
 		return;
 	}
 
-	$dir = wp_upload_dir();
-
-	$custom = file_exists( $dir['basedir'] . '/bcttstyle.css' );
+	$custom = bctt_is_custom_stylesheet();
 
 	$tag      = $custom ? 'bcct_custom_style' : 'bcct_style';
-	$antitag  = $custom ? 'bcct_style' : 'bcct_custom_style';
-	$location = $custom ? $dir['baseurl'] . '/bcttstyle.css' : plugins_url( 'assets/css/styles.css', __FILE__ );
+	$location = bctt_get_stylesheet_url();
 
 	$version = $custom ? '1.0' : '3.0';
 
 	wp_register_style( $tag, $location, false, $version, 'all' );
 
 	wp_enqueue_style( $tag );
+}
 
-	delete_option( 'bctt_style_dequeued' );
-	add_option( $tag . '_enqueued', true );
-	delete_option( $antitag . '_enqueued' );
 
+add_action( 'wp_enqueue_scripts', 'bctt_scripts', 10 );
 
+/**
+ * Check if default stylesheet must not be enqueued
+ *
+ * @return bool
+ */
+function bctt_is_default_styles_dequeued() {
+	return (bool) get_option( 'bctt_disable_css' );
 }
 
 
-add_action( 'wp_enqueue_scripts', 'bctt_scripts', 10 );
+/**
+ * Check if there's a custo stylesheet that will be enqueued
+ */
+function bctt_is_custom_stylesheet() {
+	return file_exists( bctt_get_custom_styles_path() );
```

Comment:
@joehoyle The PR did not change anything about that. It's still using `file_exists` as it was before: https://github.com/Benunc/better-click-to-tweet/blob/master/better-click-to-tweet.php#L228

The PR just tries to check if the custom stylesheet exists without checking options but there's no option that was saving the stylesheet path, they were more used as flags.

The plugin does not upload anything (at least I haven't found anything), I'm pretty sure that the file check is for a premium addon, which will upload the file in there.

I would actually use a completely different method: Use the stylesheet in a folder inside the theme, like `better-click-to-tweet/style.css` and also add some hooks to change the location through a plugin. But that would be for another PR.